src/nextjournal/clerk/analyzer.clj

```diff
@@ -243,14 +248,14 @@
                                          (and doc? (not (contains? state :ns))) (merge (parser/->doc-settings form) {:ns *ns*}))]
                              (when-let [missing-dep (and (:ns? state)
                                                          (first (set/difference (into #{}
-                                                                                      (filter #(and (symbol? %)
-                                                                                                    (#{(-> state :ns ns-name name)} (namespace %))))
+                                                                                      (comp (filter #(and (symbol? %)
+                                                                                                     (#{(-> state :ns ns-name name)} (namespace %))))
+                                                                                            (remove clojure-internal?))
```

Comment:
I felt more at ease with a fix, specific for the proxy case. Instead of any names with a dollar inside. Following the pattern of [`proxy-name`](https://github.com/clojure/clojure/blob/5ffe3833508495ca7c635d47ad7a1c8b820eab76/src/clj/clojure/core_proxy.clj#L37-L46). We can add to the predicate if we encounter more cases like this.